ibm/resource_ibm_iam_access_group_policy.go

```diff
@@ -385,12 +386,14 @@ func resourceIBMIAMAccessGroupPolicyExists(d *schema.ResourceData, meta interfac
 		accessGroupPolicyId,
 	)
 
-	accessGroupPolicy, res, err := iamPolicyManagementClient.GetPolicy(getPolicyOptions)
+	accessGroupPolicy, _, err := iamPolicyManagementClient.GetPolicy(getPolicyOptions)
 	if err != nil {
-		if res != nil && res.StatusCode == 404 {
-			return false, nil
+		if apiErr, ok := err.(bmxerror.RequestFailure); ok {
+			if apiErr.StatusCode() == 404 || *accessGroupPolicy.State == "deleted" {
```

Comment:
1. Please remove bluemix-go dependancies and also perform nil checks to avoid terraform crash
2. having deleted check inside err!=nil block will not satisfy the condition.. because I believe when err is not nil we dont get any response 


Can we add something like this
```
authorizationPolicy, resp, err := iampapClient.GetPolicy(getPolicyOptions)
if err != nil || authorizationPolicy ==nil {
        if resp !=nil && resp.StatusCode == 404 {
             return false, nil
        }
        return false, fmt.Errorf("Error communicating with the API: %s\n%s", err, resp)
}
if authorizationPolicy != nil && authorizationPolicy.State != nil && *authorizationPolicy.State == "deleted" {
  return false, nil
}
    
```
This change applies to all policy resources

Reply:
1. Removed dependency of bluemix-go in all policy resources
2. Refactored logic based on proposal
3. Changed plan type from `standard` to `lite` in `resource_ibm_iam_authorization.policy_test.go` so that a paid account is not needed for testing.